src/main/java/org/optaplanner/springboottaskassigning/solver/DefaultSolverManager.java

```diff
@@ -35,37 +37,57 @@
 
     public static final String SOLVER_CONFIG = "org/optaplanner/springboottaskassigning/solver/taskAssigningSolverConfig.xml";
 
-    private ExecutorService executorService;
+    private ExecutorService solverExecutorService;
+    private ExecutorService eventHandlerExecutorService;
     private SolverFactory<Solution_> solverFactory;
     private Map<Object, SolverTask<Solution_>> problemIdToSolverTaskMap;
+    private Map<Object, CompletableFuture<Solution_>> problemIdToCompletableFutureMap;
 
     // TODO SolverManager should be SOLVER_CONFIG agnostic, it should take the solver configuration as a constructor argument
     // TODO i.e. InputStream/File ...
     public DefaultSolverManager() {
         solverFactory = SolverFactory.createFromXmlResource(SOLVER_CONFIG, DefaultSolverManager.class.getClassLoader());
-        problemIdToSolverTaskMap = new HashMap<>();
+        problemIdToSolverTaskMap = new ConcurrentHashMap<>();
```

Comment:
Yes, because we're only adding problems/solvers and not removing any, we should probably add `boolean remove(problemId)` to the API as well, then the corresponding items should be removed from the maps.

Reply:
Why not removing the item when you call `stopSolver()`?

Reply:
Because users might want to still have access to the solver's metadata even though the solver has stopped solving (for example to allow other solvers to run in a limited resources environment). `stopSolver` semantically does not indicate that the data will be removed. So if the data is to be removed the method should be called `removeSolver`.
`problemIdToCompletableFutureMap` isn't needed any more since exceptions in the Solver thread are handled with `solverFuture.handle(...)`